docs/capacity-planning.rst

```diff
@@ -0,0 +1,326 @@
+.. _ksql_capacity_planning:
+
+======================
+KSQL Capacity Planning
+======================
+
+.. contents::
+
+KSQL is a simple and powerful tool for building streaming applications on top of Kafka. This guide will help you get started with planning for provisioning your KSQL deployment by helping you answer questions like:
+
+    - What server specification should I use to run KSQL?
+    - Roughly how many KSQL server nodes will I need?
+    - Do I need to provision additional Kafka brokers to support KSQL?
+    - What kind of deployment mode makes sense for me?
+    - How do I know whether my KSQL queries are handling the incoming message rate, and how can I tune KSQL if it’s falling behind?
+
+Background
+==========
+
+Before jumping into the following sections, you may find it instructive to look through the `Kafka Streams capacity planning guide <https://docs.confluent.io/current/streams/sizing.html>`_. Under the hood KSQL uses the `Kafka Streams API <https://docs.confluent.io/current/streams/developer-guide/dsl-api.html>`_  to perform the actual query processing, so the details covered there apply to KSQL as well.
+
+Approach To Sizing
+==================
+
+This guide will help you get a rough idea of the computing resources required to run your KSQL queries. The emphasis here is that it’s a rough idea. As you’ll see in the following sections there are many factors at play and the best practice is always to stage and test your queries against realistic load first, and adjust your planned provisioning accordingly.
+
+General Guidelines
+==================
+
+KSQL
+----
+
+**CPU**: KSQL consumes CPU to serialize/deserialize messages into the declared stream/table schemas, and then process each message as required by the query. We would recommend at least 4 cores.
```

Comment:
avoid using "we" (see [Style Guide](https://confluentinc.atlassian.net/wiki/spaces/DOC/pages/161743785/Style+Guide+for+Confluent+Documentation)). suggest:

```
At least 4 cores are recommended.
```
or 
```
You should have at least 4 cores.
```